sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/ObjectMapperFactory.java

```diff
@@ -0,0 +1,162 @@
+                .invoke();
+            useReflectionToSetCoercion = true;
+        } catch (Throwable ex) {
+            logger.verbose("Failed to retrieve MethodHandles used to set coercion configurations. "
+                + "Setting coercion configurations will be skipped.", ex);
+        }
+    }
+
+    public ObjectMapper createJsonMapper(ObjectMapperShim simpleMapperShim) {
+        ObjectMapper simpleMapper = simpleMapperShim.getMapper();
+        ObjectMapper flatteningMapper = initializeMapperBuilder(JsonMapper.builder())
+            .addModule(FlatteningSerializer.getModule(simpleMapper))
+            .addModule(FlatteningDeserializer.getModule(simpleMapper))
+            .build();
+
+        return initializeMapperBuilder(JsonMapper.builder())
+            // Order matters: must register in reverse order of hierarchy
+            .addModule(AdditionalPropertiesSerializer.getModule(flatteningMapper))
+            .addModule(AdditionalPropertiesDeserializer.getModule(flatteningMapper))
+            .addModule(FlatteningSerializer.getModule(simpleMapper))
+            .addModule(FlatteningDeserializer.getModule(simpleMapper))
+            .build();
+    }
+
+    public ObjectMapper createXmlMapper() {
+        ObjectMapper xmlMapper = initializeMapperBuilder(XmlMapper.builder())
+            .defaultUseWrapper(false)
+            .enable(ToXmlGenerator.Feature.WRITE_XML_DECLARATION)
+            /*
+             * In Jackson 2.12 the default value of this feature changed from true to false.
+             * https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.12#xml-module
+             */
+            .enable(FromXmlParser.Feature.EMPTY_ELEMENT_AS_NULL)
+            .build();
+
+
+        if (useReflectionToSetCoercion) {
+            try {
+                Object object = coersionConfigDefaults.invoke(xmlMapper);
+                setCoercion.invoke(object, coercionInputShapeEmptyString, coercionActionAsNull);
+            } catch (Throwable e) {
+                logger.verbose("Failed to set coercion actions.", e);
+            }
+        } else {
+            logger.verbose("Didn't set coercion defaults as it wasn't found on the classpath.");
+        }
+
+        return xmlMapper;
+    }
+
+    public ObjectMapper createSimpleMapper() {
+        return initializeMapperBuilder(JsonMapper.builder()).build();
+    }
+
+    public ObjectMapper createDefaultMapper() {
+        return new ObjectMapper();
+    }
+
+    public ObjectMapper createPrettyPrintMapper() {
+        return new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    }
+
+    public ObjectMapper createHeaderMapper() {
+        return initializeMapperBuilder(JsonMapper.builder())
+            .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES)
+            .build();
+    }
+
+    @SuppressWarnings("deprecation")
+    private  <S extends MapperBuilder<?, ?>> S initializeMapperBuilder(S mapper) {
```

Comment:
> nit: `MapperBuilder` isn't well supported in earlier versions of Jackson.

It supported since 2.10 and that's what we currently have in `JacksonAdapter`, i.e. we de-facto don't support 2.9 and earlier. I believe we should not start supporting them.